core/services/registrysyncer/syncer.go

```diff
@@ -20,9 +21,22 @@ type HashedCapabilityID [32]byte
 type DonID uint32
 
 type State struct {
-	IDsToDONs         map[DonID]kcr.CapabilitiesRegistryDONInfo
-	IDsToNodes        map[p2ptypes.PeerID]kcr.CapabilitiesRegistryNodeInfo
-	IDsToCapabilities map[HashedCapabilityID]kcr.CapabilitiesRegistryCapabilityInfo
+	IDsToDONs         map[DonID]kcr.CapabilitiesRegistryDONInfo                     `json:"IDsToDONs"`
+	IDsToNodes        map[p2ptypes.PeerID]kcr.CapabilitiesRegistryNodeInfo          `json:"IDsToNodes"`
+	IDsToCapabilities map[HashedCapabilityID]kcr.CapabilitiesRegistryCapabilityInfo `json:"IDsToCapabilities"`
+}
+
+func (t *State) UnmarshalJSON(input []byte) error {
```

Comment:
@vyzaldysanchez Do you need these functions? It doesn't look like they are doing anything, or have I misunderstood?

Also two suggestions for extensibility/maintainability is to:
- use two structs: one for what you return to users, and another to represent how the data is stored. That will allow you to use this layer to make any changes to the user-facing representation without affecting the stored representation
- as an extension of the above, store the version of the format somewhere -- either as part of the JSON itself, or as part of the record